src/MatrixTypes_Native.f90

```diff
@@ -801,5 +813,40 @@ SUBROUTINE transpose_SparseMatrixType(matrix)
       DEALLOCATE(n_Row)
       DEALLOCATE(Row)
     ENDSUBROUTINE transpose_SparseMatrixType
-    
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_SparseMatrixType(matrix)
+      CLASS(SparseMatrixType),INTENT(INOUT) :: matrix
+    ENDSUBROUTINE zeroentries_SparseMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_DenseSquareMatrixType(matrix)
+      CLASS(DenseSquareMatrixType),INTENT(INOUT) :: matrix
+    ENDSUBROUTINE zeroentries_DenseSquareMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_TriDiagMatrixType(matrix)
+      CLASS(TriDiagMatrixType),INTENT(INOUT) :: matrix
```

Comment:
Implement and test

Reply:
I don't believe for now there's a need for implementing this. Since the problem is only with PETSC.